### ISSUE TITLE
Hyperparameter tuning on EBM

### ISSUE THREAD

**Issue by ditskihh** [interpretml/interpret#359]
1. How can i cherry pick two pairs of specific variables in interactions parameter? from what i know the model will specify the interaction itself.
2. How can i specifically tuning on boosting model for each variables? i have concerns of some boosting model on several variables might have different parameters to have optimum score(from what i know, the parameter on EBM would apply to all function)

**Comment by paulbkoch** [comment]
Hi @ditskihh -- The interaction parameter will accept specific interactions.  You can set it like this:

```
ebm=ExplainableBoostingRegressor(interactions=[(1,2), (3, 7), (1, 11)])
ebm.fit(...)
```

We do not currently support any tuning parameters that can be set on specific features.  learning_rate might be interesting to specify per-feature, but to expose that cleanly we'd have to add a new pair specific learning rate parameter since presumably the learning_rate parameter would then be a list of length matching the number of features.  Before adding new parameters like that, we'd like to get a sense for how useful they would be.  If you wanted to hack this in temporarily to try it, a list of learning rates accessed here would do the trick:

https://github.com/interpretml/interpret/blob/34e5ddc9a525ca0968ade2908d94d466abf4d8e3/python/interpret-core/interpret/glassbox/ebm/utils.py#L1153

You can also change the max_leaves and min_samples_leaf there.
